sandbox/reflection/test.cc

```diff
@@ -0,0 +1,20 @@
+#include "reflection.h"
+
+#include "../../3rdparty/gtest/gtest-main.h"
+
+namespace reflection_test {
+CURRENT_STRUCT(Foo) { CURRENT_FIELD(uint64_t, i); };
+static_assert(CURRENT_STRUCT_IS_VALID(Foo), "Struct `Foo` was not properly declared.");
+}  // namespace reflection_test
+static_assert(CURRENT_STRUCT_IS_VALID(reflection_test::Foo), "Struct `Foo` was not properly declared.");
```

Comment:
:+1:

Perhaps also check `CURRENT_STRUCT_IS_VALID` from another namespace and/or from within another top-level class? :-)